src/metaopt/worker/trial.py

```diff
@@ -0,0 +1,235 @@
+                type=self._type,
+                value=self.value
+                )
+            return ret
+
+        def __str__(self):
+            """Represent partially with a string."""
+            ret = "{0}(name={1}, type={2}, value={3})".format(
+                type(self).__name__, repr(self.name),
+                repr(self._type), repr(self.value))
+            return ret
+
+        __repr__ = __str__
+
+        @property
+        def type(self):
+            """For meaning of property type, see `Value.type`."""
+            return self._type
+
+        @type.setter
+        def type(self, type_):
+            if type_ is not None and type_ not in self.allowed_types:
+                raise ValueError("Given type, {0}, not one of: {1}".format(
+                    type_, self.allowed_types))
+            self._type = type_
+
+    class Result(Value):
+        """Types for a `Result` can be either an evaluation of an 'objective'
+        function or of an 'constaint' expression.
+        """
+
+        allowed_types = ('objective', 'constraint')
+
+    class Param(Value):
+        """Types for a `Param` can be either an integer (discrete value),
+        floating precision numerical or a categorical expression (e.g. a string).
+        """
+
+        allowed_types = ('int', 'float', 'enum')
+
+    __slots__ = ('experiment', '_id', '_status', 'worker',
+                 'submit_time', 'start_time', 'end_time', 'results', 'params')
+    allowed_stati = ('new', 'reserved', 'suspended', 'completed', 'broken')
+
+    def __init__(self, **kwargs):
+        """See attributes of `Trial` for meaning and possible arguments for `kwargs`."""
+        for attrname in self.__slots__:
+            if attrname in ('results', 'params'):
+                setattr(self, attrname, list())
+            elif attrname == '_id':
+                continue
+            else:
+                setattr(self, attrname, None)
+
+        self.status = 'new'
+
+        for attrname, value in six.iteritems(kwargs):
+            if attrname == 'results':
+                attr = getattr(self, attrname)
+                for item in value:
+                    attr.append(self.Result(**item))
+            elif attrname == 'params':
+                attr = getattr(self, attrname)
+                for item in value:
+                    attr.append(self.Param(**item))
+            else:
+                setattr(self, attrname, value)
+
+    def to_dict(self):
+        """Needed to be able to convert `Trial` to `dict` form."""
+        ret = dict(
+            status=self._status,
+            results=list(map(lambda x: x.to_dict(), self.results)),
+            params=list(map(lambda x: x.to_dict(), self.params))
+            )
+        for attrname in self.__slots__:
+            if attrname in ('results', 'params', '_status'):
+                continue
+            try:
+                ret[attrname] = getattr(self, attrname)
+            except AttributeError:
```

Comment:
I didn't realized that for _id. Then,

```python
    # Trial object should not be able to set a value
    # (whatever value) to its `_id` on its own, but is should just
    # expect that it could have one
    trial_dictionary.pop("id")  # pop invalid attrname
    if self.is_registered():
        trial_dictionary["_id"] = self.id
```

I understand that we need to send a dict without "_id" if the Trial is not registered. That's what I'm trying to do without entangling this with other potential unset slots causing AttributeError.